src/transport/CryptoContext.cpp

```diff
@@ -147,6 +147,15 @@ CHIP_ERROR CryptoContext::BuildNonce(NonceView nonce, uint8_t securityFlags, uin
     return bbuf.Fit() ? CHIP_NO_ERROR : CHIP_ERROR_NO_MEMORY;
 }
 
+CHIP_ERROR CryptoContext::BuildPrivacyNonce(NonceView nonce, uint16_t sessionId, const MessageAuthenticationCode & mac)
+{
+    Encoding::LittleEndian::BufferWriter bbuf(nonce.data(), nonce.size());
+
+    bbuf.Put16(sessionId);
+    bbuf.Put(mac.GetTag(), kPrivacyNonceMicFragmentLen);
```

Comment:
This does not follow spec!

- The BufferWriter must be Big-Endian (the session ID must be written big-endian)
- The last 11 bytes of the MIC must be used, not the first 11.

See https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/5518

Reply:
Corrected and added unit test